exec/cnex/lib/io.c

```diff
@@ -0,0 +1,189 @@
+#include "io.h"
+#include "enums.h"
+
+#ifdef _MSC_VER
+#define _CRT_SECURE_NO_WARNINGS
+#endif
+#ifdef _WIN32
+#include <io.h>
+#endif
+#include <assert.h>
+#include <iso646.h>
+#include <stdio.h>
+#include <string.h>
+#ifndef _WIN32
+#include <unistd.h>
+#include <sys/types.h>
+#endif
+
+#include "cell.h"
+#include "exec.h"
+#include "number.h"
+#include "stack.h"
+#include "nstring.h"
+
+
+static FILE *check_file(TExecutor *exec, void *pf)
+{
+    FILE *f = (FILE *)(pf);
+    if (f == NULL) {
+        exec->rtl_raise(exec, "InvalidFile", "", BID_ZERO);
```

Comment:
The exception to be raised here is `IoException.InvalidFile`. The same applies to the other places you raise an exception in here.